### DISCUSSION TITLE
Using data from API request

### DISCUSSION THREAD

**Discussion by fabiothpita** [rappasoft/laravel-livewire-tables#1283] [General]
Hello there! I'm a newcomer to Livewire and I've come across this package. While going through the documentation, I noticed that it uses Models in its examples. I'm wondering if there's a method to populate tables with data from API requests. 

I appreciate your help in advance.

**Comment by lrljoe** [comment]
Hi Fabio,

The whole intention behind this particular package is to allow searching, filtering and sorting of models.

Sadly I can't think of a simple way to consume an API and utilise the functionality.

Interested in what your use case is though for parsing an API into a table, as there will be a good option for you

**Reply by fabiothpita** [reply]
Hello Joe,

I hope you're doing well. I wanted to thank you to taking the time to reply. I'm currently working on a project that involves an API from a legacy system, and I've chosen to utilize Livewire for crafting the user interface. While working on this, I realized the necessity of creating a table, which led me to discover your project. I wanted to integrating it into my own work.

Since it's not possible, I'll have to explore other alternatives or potentially develop a solution from scratch.

Nonetheless, I truly appreciate your assistance and insight. 😄

**Reply by lrljoe** [reply]
Certainly not my project, @rappasoft kicked this all off, I just started meddling in the last year or two.

Having worked with hundreds of legacy systems over the years, honestly, I'd get the data and turn it into models rather than rely on a legacy system.
You could relatively easily import the data.

If you're stuck with them however, then there are some AlpineJS/Tailwind alternatives such as:
https://tailwindcomponents.com/component/table-ui-with-tailwindcss-and-alpinejs

I'd definitely avoid looking exclusively for Livewire packages, as they'll almost all be exclusively Livewire & Laravel Eloquent (model) focused.

AlpineJS and either Tailwind/Bootstrap (save yourself a headache, and choose one!)  are the frontend for it all really, and Livewire just provides an inteface to the data.

It's ceratinly an interesting question though.

I'd definitely be dumping it all into models so that it's persisted, and I can query it how I want to, rather than being limited by API methods provided by a legacy system.

I did do some work with a 3rd party a while back, where they had to query an API, but we ended up using models as a type of middleware almost, as being able to freely filter/search/paginate the results made it much easier to work with the data.

To be honest, I can't imagine not persisting the data in some form locally to add functionality, in which case it may as well be an Eloquent model.

**Reply by fabiothpita** [reply]
Thanks for your opinion about my use case and for the alternative. I will try using Tailwind and AlpineJS first. If that doesn't work, I will find another alternative. I hope using this package in my future projects 😄

**Reply by lrljoe** [reply]
https://github.com/tttstudios/alpinejs-table

that will probably be your best bet

**Reply by fabiothpita** [reply]
Thanks, I appreciated
